### PR TITLE
Generates xcconfigs beased on the  configuration

### PR DESCRIPTION
**Pull request by congt**
Supports release and debug builds.

### CODE REVIEW

File: lib/cocoapods/bazel/config.rb [hunk 1]
```diff
@@ -48,6 +49,14 @@ def buildifier
         to_h[:buildifier]
       end
 
+      def configuration
```
Comment (segiddins) [inline lib/cocoapods/bazel/config.rb:52]: why is this part of the config? I feel like we should be generating build files that work for both debug & release configurations. @thiagohmcruz didn't you add an attr for `xcconfigs_by_condition` that we should be able to leverage here?

Comment (thiagohmcruz) [reply lib/cocoapods/bazel/config.rb:52]: Yep, that's how I imagined this being implemented here. @congt I think we could use the build settings being created in `Pods/cocoapods-bazel/BUILD.bazel` by `cocoapods-bazel` to create BUILD files having support for both `:release` and `:debug`

You can see one example [here](https://github.com/bazel-ios/rules_ios/blob/09c9fa466632f730fc13d2687a5f77330e909e29/tests/ios/app/BUILD.bazel#L216)

Does that makes sense?

File: spec/integration/release-build/after/Frameworks/a/BUILD.bazel [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+    ]),
+    bundle_id = "org.cocoapods.A-App",
+    families = [
+        "iphone",
+        "ipad",
+    ],
+    minimum_os_version = "10.0",
+    module_name = "A_App",
+    swift_version = "5.2",
+    xcconfig = {
```
Comment (thiagohmcruz) [inline spec/integration/release-build/after/Frameworks/a/BUILD.bazel:46]: The `after` BUILD file here would look similar to the one I mentioned before. Leveraging `xcconfig_by_build_setting `

File: lib/cocoapods/bazel/config.rb (hunk 1 above)
Comment (congt) [reply lib/cocoapods/bazel/config.rb:52]: We can put configuration specific xcconfigs in `xcconfig_by_build_setting`, but I realized that many attributes (e.g., bundle_id, entitlements, swift_copts) also depend on the configuration. For example, `bundle_id` is set with the setting `PRODUCT_BUNDLE_IDENTIFIER` which equals to `$(PRODUCT_BUNDLE_IDENTIFIER_$(CONFIGURATION))`. Given that, we will have to pass the configuration option into this plugin.

Comment (segiddins) [reply lib/cocoapods/bazel/config.rb:52]: I think those do belong in selects, otherwise targets won't really work for both configurations

Comment (congt) [reply lib/cocoapods/bazel/config.rb:52]: OK, that's a fair point. I guess we eventually need to support that any  way.

**Pull request closed without merging**